src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -483,9 +492,14 @@ bool D4NFilterObject::check_head_exists_in_cache_get_oid(const DoutPrefixProvide
           };
 
   bool found_in_cache = true;
-  int ret = -1;
+  int ret = blockDir->get(dpp, &block, y);
+  block.cacheObj.objName = this->get_name(); // account for blocks representing simple requests
   //if the block corresponding to head object does not exist in directory, implies it is not cached
-  if ((ret = blockDir->get(dpp, &block, y) == 0)) {
+  if (ret == 0 || (ret = blockDir->get(dpp, &block, y)) == 0) {
```

Comment:
You have added a second get() call here, which will impact every read call, and I dont this is acceptable. Can you justify why this is needed